src/Core/Configure.php

```diff
@@ -78,10 +79,10 @@ class Configure
      * @param string|array $config The key to write, can be a dot notation value.
      * Alternatively can be an array containing key(s) and value(s).
      * @param mixed $value Value to set for var
-     * @return bool True if write was successful
+     * @return void
      * @link https://book.cakephp.org/3.0/en/development/configuration.html#writing-configuration-data
      */
-    public static function write($config, $value = null)
+    public static function write($config, $value = null): void
```

Comment:
In the vein of #12335 this seems like a similar change that could disrupt folks more than we need to. I agree that always returning `true` is a bit goofy, but it is a consistent type at least.